libs/utils/src/postgres_backend_async.rs

```diff
@@ -4,45 +4,78 @@
 //! is rather narrow, but we can extend it once required.
 
 use crate::postgres_backend::AuthType;
-use anyhow::{bail, Context, Result};
+use anyhow::Context;
 use bytes::{Bytes, BytesMut};
 use pq_proto::{BeMessage, FeMessage, FeStartupPacket};
 use rand::Rng;
-use std::future::Future;
 use std::net::SocketAddr;
 use std::pin::Pin;
 use std::sync::Arc;
 use std::task::Poll;
-use tracing::{debug, error, trace};
+use std::{future::Future, io};
+use tracing::{debug, error, info, trace};
 
 use tokio::io::{AsyncRead, AsyncWrite, AsyncWriteExt, BufReader};
 use tokio_rustls::TlsAcceptor;
 
+#[derive(thiserror::Error, Debug)]
```

Comment:
One funny note here is that `thiserror::Error` is kind of hazardous here: due to its presence, we can accidentally write `?` somewhere and convert this error into `anyhow::Error` without noticing.

I've made such error twice and had to remove the annotations temporarily: it's possible to have just `Display` , maybe also `Debug`, but not `std::error::Error` trait for the current code to compile and for `?` to never compile.